benchmark/synthetic/models/lstm_text_generation_wikitext2.py

```diff
@@ -0,0 +1,190 @@
+'''
+Original Model from https://github.com/keras-team/keras/blob/master/examples/lstm_text_generation.py
+Benchmark for a LSTM model.
+Script template modified from TensorFlow Benchmark repo:
+https://github.com/tensorflow/benchmarks/blob/keras-benchmarks/scripts/keras_benchmarks/models/lstm_benchmark.py
+
+Used WikiText-2 character level dataset from
+https://einstein.ai/research/the-wikitext-long-term-dependency-language-modeling-dataset
+'''
+from __future__ import print_function
+import keras
+from keras.utils import multi_gpu_model
+
+from models import timehistory
+
+from keras.callbacks import LambdaCallback
+from keras.models import Sequential
+from keras.layers import Dense, Activation
+from keras.layers import LSTM
+from keras.optimizers import RMSprop
+import numpy as np
+import random
+import sys
+import io
+import re
+import os
+
+if keras.backend.backend() == 'tensorflow':
+    import tensorflow as tf
+if keras.backend.backend() == 'cntk':
+    from gpu_mode import cntk_gpu_mode_config, finalize
+
+
+def crossentropy_from_logits(y_true, y_pred):
+    return tf.nn.softmax_cross_entropy_with_logits(labels=y_true, logits=y_pred)
+
+
+class LstmBenchmark:
+
+    def __init__(self):
+        self.test_name = "lstm_text_generation_wikitext2"
+        self.sample_type = "text"
+        self.total_time = 0
+        self.batch_size = 128
+        self.epochs = 20
+
+    def run_benchmark(self, gpus=0, inference=False, use_dataset_tensors=False):
+        print("Running model ", self.test_name)
+        keras.backend.set_learning_phase(True)
+
+        dir_path = os.path.dirname(os.path.realpath(__file__))
+        path = dir_path + "/wiki.train.raw"
```

Comment:
any particular reason as to why are you downloading this dataset before? why not use the S3 path directly as you have done for nietzsche?

Reply:
The dataset is hosted by different AI research group, unlike nietzsche. So I thought it's better to give a user a link to that research group and download manually.

Reply:
You can mention that link in the ReadMe & follow the same style as the nietzsche dataset for importing it for consistency

Reply:
Now it supports automated scripts that download the dataset from S3 bucket and it also handles the cache where if a user already has that dataset then it won't download it again.